src/devices/machine/mc6852.cpp

```diff
@@ -121,7 +130,42 @@ void mc6852_device::device_reset()
 
 void mc6852_device::tra_callback()
 {
-	m_write_tx_data(transmit_register_get_data_bit());
+	int bit=0;
+	const uint8_t tx_bit_total = 8; 
+
+	int size = m_tx_fifo.size();
+	LOGTX("Transmit FIFO size %d\n", size);
+
+	if (size > 0)
+	{
+
+		uint8_t data = m_tx_fifo.front();
+
+		bit = (data>>(tx_bit_total-1-m_tx_bit_count))&1;
```

Comment:
Please use `const` when possible, use the `BIT` helper function for extracting bits, and don’t code like your space bar is broken.

Reply:
Fair feedback, will clean up.